app/user.go

```diff
@@ -981,6 +981,51 @@ func (a *App) UpdateActive(c request.CTX, user *model.User, active bool) (*model
 		}
 	}
 
+	dmChannels := []*model.Channel{}
+	channelsInTeam, aErr := a.GetChannelsForTeamForUser(c, "", user.Id, &model.ChannelSearchOpts{
+		IncludeDeleted: true,
+		LastDeleteAt:   0,
+	})
+	// soft errors here on since user is updated
+	if aErr != nil {
+		c.Logger().Error("Failed to get channels for user", mlog.String("user_id", user.Id), mlog.Err(err))
+	}
+
+	if channelsInTeam != nil {
+		for _, channel := range channelsInTeam {
+			if channel.Type == model.ChannelTypeDirect {
+				dmChannels = append(dmChannels, channel)
+			}
+		}
+		// channelMembers, err := a.GetChannelMembersForUser()
+		if active {
+			// undelete direct message channels
+			for _, channel := range dmChannels {
+				// not using app.GetChannelMembersCount because it computes from cache
+				channelMemberCount, err := a.Srv().Store().Channel().GetMemberCount(channel.Id, false)
```

Comment:
Are there cases where DM channels have more than 2 members?

Reply:
It's a `model.ChannelTypeGroup` if it has multiple members, and there's a condition above to only filter DM channels.